src/hera/workflows/io/_io_mixins.py

```diff
@@ -1,7 +1,27 @@
+import sys
 import warnings
 from pathlib import Path
 from typing import TYPE_CHECKING, List, Optional, Union
 
+if sys.version_info >= (3, 9):
+    from typing import Annotated
+else:
+    from typing_extensions import Annotated
+
+if sys.version_info >= (3, 10):
+    from typing import get_args, get_origin
+else:
+    # Python 3.8 has get_origin() and get_args() but those implementations aren't
+    # Annotated-aware. Python 3.9's versions don't support ParamSpecArgs and
+    # ParamSpecKwargs.
+    from typing_extensions import get_args, get_origin
```

Comment:
The old version of this code was importing `Self` in the try/except block, causing it to fall back to the bespoke implementation in `typing_extensions` for Python 3.9. I'm unclear if this was intentional, so retained this behaviour to avoid any potential regression. For Python 3.10, `typing_extensions` imports Python's built-in version, so we can import it directly.